### PR TITLE
Update cilium documentation for 1.6 update

### PR DESCRIPTION
**Pull request by soumynathan**
This patch updates architecture-description.adoc with cilium new
features for 1.6 release

Signed-off-by: Swaminathan Vasudevan <svasudevan@suse.com>

### CODE REVIEW

File: adoc/architecture-description.adoc [hunk 1]
```diff
@@ -201,6 +201,106 @@ consumes this service, the kube proxy is the responsible for setting
...
++
+The addition of a new CRD-based backend for security identities now allows to
+operate Cilium entirely without a KVstore in the context of Kubernetes.
+
+* Cilium 100% Kube-proxy replacement
++
+Operating a Kubernetes cluster without requiring to run kube-proxy has been a
+desire of many users. This release Cilium 1.6 includes the final two features
+required to run a Kubernetes cluster without kube-proxy with Cilium fully
+replacing kube-proxy.
```
Comment (manuelbuil) [inline adoc/architecture-description.adoc:270]: Careful, we haven't done the kube-proxy replacement in CaaSP. We will wait for SLE 15 SP2

File: adoc/architecture-description.adoc [hunk 2]
```diff
@@ -201,6 +201,106 @@ consumes this service, the kube proxy is the responsible for setting
...
+Envoy is primarily designed as a service to service communication system, but
+there is benefit in using the same software at the edge (observability,
+management, identical service discovery and load balancing algorithms, etc.).
+Envoy includes enough features to make it usable as an edge proxy for most
+modern web application use cases. This includes TLS termination, HTTP/1.1 and
+HTTP/2 support, as well as HTTP L7 routing.
+
+The primary goal of Envoy is to make the network transparent.
+
+Cilium 1.6 supports Envoy proxy.
```
Comment (manuelbuil) [inline adoc/architecture-description.adoc:258]: I'd say that Cilium 1.6 uses Envoy for the L7 Network Policies

File: adoc/architecture-description.adoc (hunk 1 above)
Comment (soumynathan) [reply adoc/architecture-description.adoc]: Done.

File: adoc/architecture-description.adoc (hunk 2 above)
Comment (soumynathan) [reply adoc/architecture-description.adoc]: Done

**Comment by manuelbuil** [comment]
LGTM

**Comment by r0ckarong** [comment]
@soumynathan 

I can not use this. The text is verbatim copy+paste from the Envoy docs and Cilium release notes. That would be plagiarism and copyright theft.

Please write (in your own words) a short introduction into envoy (doesn't have to be more than 2-3 paragraphs long and talk about the very basics, then add a link to the "what is envoy" page you took this text from.

Same for Cilium, please remove the verbatim bullet points from the 1.6 release notes and write a short text about why cilium and then you can link to the features in the release notes.

We could alternatively rework this entire page into a mess of quotation blocks but that would look terrible and won't be very readable.

**Comment by soumynathan** [comment]
> @soumynathan
> 
> I can not use this. The text is verbatim copy+paste from the Envoy docs and Cilium release notes. That would be plagiarism and copyright infringement with (utlimately) my name on it.
> 
> Please write (in your own words) a short introduction into envoy (doesn't have to be more than 2-3 paragraphs long) and talk about the very basics and important things. Then it's fine to add a link to the "what is envoy" page you took this text from.
> 
> Same for Cilium, please remove the verbatim bullet points from the 1.6 release notes and write a short text about why cilium and then you can link to the features in the release notes.
> 
> We could alternatively rework this entire page into a mess of quotation blocks but that would look terrible and won't be very readable.

Sure will address this. I was not sure about leveraging the documentation.

**Comment by r0ckarong** [comment]
Much better, thank you!

**Pull request merged**